xcdat/temporal.py

```diff
@@ -919,10 +919,28 @@ def _group_average(self, data_var: xr.DataArray) -> xr.DataArray:
 
         if self._weighted:
             self._weights = self._get_weights()
+            # grab metadata
+            dv_attrs = dv.attrs
+            dv_name = dv.name
+            # grab metadata (populated after calling ._group_data())
+            lt_attrs = self._labeled_time.attrs
+            lt_encoding = self._labeled_time.encoding
+            # weight the data variable
             dv *= self._weights
-            dv = self._group_data(dv).sum()
+            # cast the weights to match the dv.shape / dims
+            weights, x = xr.broadcast(self._weights, dv)
```

Comment:
`weights` is one-dimensional (with a temporal dimension), but I want to make sure that grid cells that are `np.nan` in `dv` receive zero weight. Since `dv` is multidimensional, I need my `weights` array to have the same shape in order to reflect this.

Reply:
Incorporated this PR comment in the code comment.